From c50d5574e88e0ecc55b720c6a494fc42a51de71d Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:23 -0400
Subject: [infiniband] mlx4_ib: Adapt code to use caps.num_ports instead of a constant

Message-id: <47e542769acdeb569c712675a1f5bf013337d853.1410425016.git.avadai@redhat.com>
Patchwork-id: 93756
O-Subject: [Patch RHEL7.1 V1 087/197] IB/mlx4_ib: Adapt code to use caps.num_ports instead of a constant
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Matan Barak <matanb@mellanox.com>

Some code in the mlx4 IB driver stack assumed MLX4_MAX_PORTS ports.

Instead, we should only loop until the number of actual ports in i
the device, which is stored in dev->caps.num_ports.

Signed-off-by: Matan Barak <matanb@mellanox.com>
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 82373701be26b893eaf7372db0af84235a51998a)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/main.c b/drivers/infiniband/hw/mlx4/main.c
index 6fa0fe5..d8bd933 100644
--- a/drivers/infiniband/hw/mlx4/main.c
+++ b/drivers/infiniband/hw/mlx4/main.c
@@ -1546,7 +1546,7 @@ static int mlx4_ib_addr_event(int event, struct net_device *event_netdev,
  iboe = &ibdev->iboe;
  spin_lock(&iboe->lock);
 
- for (port = 1; port <= MLX4_MAX_PORTS; ++port)
+ for (port = 1; port <= ibdev->dev->caps.num_ports; ++port)
   if ((netif_is_bond_master(real_dev) &&
        (real_dev == iboe->masters[port - 1])) ||
        (!netif_is_bond_master(real_dev) &&
@@ -1569,14 +1569,14 @@ static u8 mlx4_ib_get_dev_port(struct net_device *dev,
 
  iboe = &ibdev->iboe;
 
- for (port = 1; port <= MLX4_MAX_PORTS; ++port)
+ for (port = 1; port <= ibdev->dev->caps.num_ports; ++port)
   if ((netif_is_bond_master(real_dev) &&
        (real_dev == iboe->masters[port - 1])) ||
        (!netif_is_bond_master(real_dev) &&
        (real_dev == iboe->netdevs[port - 1])))
    break;
 
- if ((port == 0) || (port > MLX4_MAX_PORTS))
+ if ((port == 0) || (port > ibdev->dev->caps.num_ports))
   return 0;
  else
   return port;
@@ -1626,7 +1626,7 @@ static void mlx4_ib_get_dev_addr(struct net_device *dev,
  union ib_gid gid;
 
 
- if ((port == 0) || (port > MLX4_MAX_PORTS))
+ if ((port == 0) || (port > ibdev->dev->caps.num_ports))
   return;
 
  /* IPv4 gids */
-- 
1.7.1